src/components/ConfigEditor/ConnectionConfig.tsx

```diff
@@ -164,33 +165,36 @@ const ConnectionConfig: React.FC<ConnectionConfigProps> = ({
         tooltip={clientSecretTooltip}
       />
 
-      <InlineField
-        label="Use On-Behalf-Of"
-        labelWidth={26}
-        tooltip={
-          <>
-            Propagate Grafana client credentials to ADX with a token exchange. When enabled the service account (Client
-            ID) impersonates the user by augmenting the access token. See the{' '}
-            <a
-              target="_blank"
-              rel="noreferrer"
-              href="https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow"
-            >
-              developer documentation
-            </a>{' '}
-            for detail on the concept. This feature requires a Grafana version 8.3.4 or later.
-          </>
-        }
-      >
-        <HorizontalGroup>
-          <InlineSwitch
-            id="adx-on-behalf-of"
-            value={jsonData.onBehalfOf}
-            onChange={(ev: React.ChangeEvent<HTMLInputElement>) => updateJsonData('onBehalfOf', ev.target.checked)}
-          />
-          <span>⚠️ This feature is in beta</span>
-        </HorizontalGroup>
-      </InlineField>
+      {config.featureToggles.adxOnBehalfOf && (
+        <InlineField
+          label="Use On-Behalf-Of"
+          htmlFor="adx-on-behalf-of"
```

Comment:
Adding `htmlFor` explicitly as the `getChildId` used in `InlineField` will only look at first level children (i.e.`HorizontalGroup` here).

Having a correct `for` wiring helps with RTL and a11y 🚀